core/vm/actor/builtin/types/market/deal.hpp

```diff
@@ -119,7 +59,7 @@ namespace fc::vm::actor::builtin::types::market {
   CBOR_TUPLE(DealState, sector_start_epoch, last_updated_epoch, slash_epoch)
 
   struct ClientDealProposal {
-    DealProposal proposal;
+    Universal<DealProposal> proposal;
```

Comment:
`proposal` is ptr now and default copy assigment of `ClientDealProposal` is a shallow copy.
Please, take a look.

Reply:
If you need a full copy, please write copy constructor and operator for `ClientDealProposal` where call `Universal::copy`